File: app/src/main/java/sakaitakao/android/permissionsearch/dialog/SaveAdvancedSearchConditionDialog.java
package sakaitakao.android.permissionsearch.dialog;

import sakaitakao.android.permissionsearch.R;
import android.app.Dialog;
import android.content.Context;
import android.widget.Button;

/**
 * 詳細検索の検索条件の名前を入力するためのダイアログ
 * 
 * @author takao
 * 
 */
public class SaveAdvancedSearchConditionDialog extends Dialog {

	/**
	 * ダイアログを生成します。
	 * 
	 * @param context
	 */
	public SaveAdvancedSearchConditionDialog(Context context) {
		super(context);

		setContentView(R.layout.dlg_save_condition);
		setTitle(R.string.title_dlg_save_condition);
	}

	/**
	 * OK ボタンが押されたら呼び出すリスナを設定する。
	 * 
	 * @param onClickListener
	 */
	public void setOkButtonOnClickListener(android.view.View.OnClickListener onClickListener) {
		Button buttonOk = (Button) findViewById(R.id.dlg_save_condition_ok);
		buttonOk.setOnClickListener(onClickListener);
	}
}

File: app/src/main/kotlin/sakaitakao/android/permissionsearch/dialog/SaveAdvancedSearchConditionDialog.kt
package sakaitakao.android.permissionsearch.dialog

import android.app.Dialog
import android.content.Context
import android.widget.Button
import sakaitakao.android.permissionsearch.R

/**
 * 詳細検索の検索条件の名前を入力するためのダイアログ

 * @author takao
 */
class SaveAdvancedSearchConditionDialog
/**
 * ダイアログを生成します。

 * @param context
 */
(context: Context) : Dialog(context) {

    init {

        setContentView(R.layout.dlg_save_condition)
        setTitle(R.string.title_dlg_save_condition)
    }

    /**
     * OK ボタンが押されたら呼び出すリスナを設定する。

     * @param onClickListener
     */
    fun setOkButtonOnClickListener(onClickListener: android.view.View.OnClickListener) {
        val buttonOk = findViewById(R.id.dlg_save_condition_ok) as Button
        buttonOk.setOnClickListener(onClickListener)
    }
}
